internal/cephfs/controllerserver.go

```diff
@@ -36,25 +40,81 @@ type ControllerServer struct {
 	// A map storing all volumes with ongoing operations so that additional operations
 	// for that same volume (as defined by VolumeID/volume name) return an Aborted error
 	VolumeLocks *util.VolumeLocks
+
+	// A map storing all snapshots with ongoing operations so that additional operations
+	// for that same snapshot (as defined by SnapshotID/snapshot name) return an Aborted error
+	SnapshotLocks *util.VolumeLocks
 }
 
 // createBackingVolume creates the backing subvolume and on any error cleans up any created entities.
-func (cs *ControllerServer) createBackingVolume(ctx context.Context, volOptions *volumeOptions, vID *volumeIdentifier, secret map[string]string) error {
-	cr, err := util.NewAdminCredentials(secret)
-	if err != nil {
-		return status.Error(codes.InvalidArgument, err.Error())
+func (cs *ControllerServer) createBackingVolume(
```

Comment:
This function seems to return CSI-errors, or ErrCloneInProgress and maybe others (from `createCloneFromSubvolume()`). It would be cleaner to return CSI-errors in all cases, so that the caller of `createBackingVolume()` does not need to check and do additional conversion of the error.

In case the non CSI-errors are needed elsewhere, document the different errors that can be returned.

Reply:
@nixpanic  As discussed this is added as a cleanup or enhancement in the tracker , will pick up as soon as this is in. thanks !